roles/sql-command/tasks/mysql-query.yml

```diff
@@ -1,28 +1,46 @@
 ---
-
 - name: Create my.cnf file
   include: create-my-cnf.yml
 
-- name: prepare query from template
-  template:
-    src: "templates/ad-hoc-query.sql"
-    dest: "/home/{{ ssh_prov_user|default('jenkins-agent') }}/ad-hoc-query.sql"
-    force: yes
-  delegate_to: "master.{{ vpc_name }}.internal"
-  no_log: true
+- block:
+    - name: prepare query from template
+      template:
+        src: "templates/ad-hoc-query.sql"
+        dest: "/home/{{ ssh_prov_user|default('jenkins-agent') }}/ad-hoc-query.sql"
+        force: yes
+      delegate_to: "{{ 'master.' + vpc_name +'.internal' }}"
+      no_log: true
+
+    - name: Execute our sql command
+      shell: "mysql --defaults-file=/home/{{ ssh_prov_user|default('jenkins-agent') }}/my.cnf
+            -h {{ sql_host }}.internal
+            {{ target_db_name }} < /home/{{ ssh_prov_user|default('jenkins-agent') }}/ad-hoc-query.sql"
+      delegate_to: "{{ 'master.' + vpc_name +'.internal' }}"
+      register: mysql_result
+      no_log: true
+
+    - name: remove sql template
+      file:
+        dest: "/home/{{ ssh_prov_user|default('jenkins-agent') }}/ad-hoc-query.sql"
+        state: absent
+      delegate_to: "{{ 'master.' + vpc_name +'.internal' }}"
+  when: remote_file is not defined
 
-- name: Execute our sql command
-  shell: "mysql -h {{ rds_instance_data.private_dns + '.' + opg_data.stack }}.internal
-        {{ rds_instance_data.db_name }} < /home/{{ ssh_prov_user|default('jenkins-agent') }}/ad-hoc-query.sql"
-  delegate_to: "master.{{ vpc_name }}.internal"
-  register: mysql_result
-  no_log: true
+- block:
+    - name: Execute our sql command
```

Comment:
`when: remote_file is defined` is the control statement for this block, it will execute a sql file on the master that already exists, either generated by an ansible play or possibly something persistent.

In this case it's used to do a mysql restore from the data dump in the repository. See https://github.com/ministryofjustice/opg-ansible-roles/pull/64/files/bbe225e7168d68513292229cc0aede4dc081d40e#diff-7d37bb508b0be77edb4dc4b79394e661